### PR TITLE
Basics of JS-IPFS

### LINKED ISSUE #1207: Create an section covering the Basics of IPFS using JavaScript.
This is a continuation of the _Basics of IPFS_ project, where we show users how to use IPFS in very basic ways (install, add, share/get, and _"remove"_). In this new subsection, we'll show the reader how to spin up an IPFS node in the browser using JavaScript and then, in subsequent tabs, add a file, share that file with another user, how the retrieve a file, and finally, how to _"remove"_ a file from the JS-IPFS node. This new section can follow the same formula as used in the current [IPFS Basics section](https://docs.ipfs.io/basics/).

## Tasks

Here's a pretty rough overview of what we should cover in this section.

- [x] How to _install_ JS-IPFS.
    - [x] Where to get JS-IPFS from: https://github.com/ipfs/js-ipfs
    - [x] Who it's maintained by.
    - [x] Show the simple `npm install ...` procedure.
    - [x] Also mention how a user can build from source if they want to.
        - [x] But just point them to the instructions in the JS-IPFS repo.
- [x] Spin up an IPFS node
    - [x] Build a simple webpage (without the use of any frameworks like React/Angular/etc.).
    - [x] Show the user the code needed to spin up a node.
    - [x] Have the JS Console output something once the node is running.
    - [x] Explain that all the other steps will use this code.
- [ ] Add a file.
    - [x] Copy the code from the _Spin up an IPFS node_ section. (not entirely based on this)
    - [ ] Add a dialog box where a user can put the file they want to upload.
    - [x] Once the user clicks upload, the file is sent to the JS-IPFS node.
    - [x] Output the CID on the webpage.
    - [x] Mention that they can then share this CID with anyone, and they can use it on their IPFS node to get that file.
- [ ] Get a CID.
    - [x] Copy the code from the _Spin up an IPFS node_ section. (not entirely based on this)
    - [ ] Create an input box where a user can paste a CID.
    - [x] Have the user enter a specific CID that we put somewhere and have pinned. A simple jpeg image would work well.
    - [x] Once the user requests that CID, try to show it in the browser.
- [ ] Remove a file.
    - [x] Start off by explaining that it's not possible to 100% verifiable remove a file/CID from the IPFS network.
        - [x] There's no way to know if someone else had made a copy of the file since it's been on the network.
        - [x] This works the same way as it does in regular HTTP networks, etc.
    - [x] Copy the code from the _Spin up an IPFS node_ section. (not entirely based on this)
    - [x] List the CIDs that this node is currently hosting.
    - [x] Add some kind of user input that the user can use to delete a file from this JS-IPFS node.
    - [x] Once the file has been removed, relist the CIDs that this node is currently hosting.

### PR DESCRIPTION
**Pull request by DannyS03**
Closes #1207

- adds steps as proposed in issue. leaves out the more interactive pieces (such as dialog and input box) 
- proposes the use of tabs: using an external plugin which can be changed
- modifies and updates JS content

### CODE REVIEW

**Comment by johnnymatthews** [comment]
This looks fun! There's some weird stuff going on with the 

https://user-images.githubusercontent.com/9611008/182938261-933057c8-89ac-4e82-b154-8c2c07a7d450.mov

tabs though. Clicking on one of the tabs under **Install JS-IPFS** jumps me down to the next section:

File: docs/basics/go/command-line.md [hunk 1]
```diff
@@ -1,11 +1,11 @@
 ---
-title: "Command-line"
+title: IPFS in Golang
 description: "A simple walkthrough of how to perform basic IPFS operations using the Kubo command-line."
 ---
 
 # Basic CLI Operations
 
-This short guide aims to walk you through the basics of using IPFS with the Kubo CLI.  Kubo is [one of multiple IPFS implementations](.ipfs-implementations.md). It is the oldest IPFS implementation and exposes a CLI (among other things).
+This short guide aims to walk you through the **basics of using IPFS with the Kubo CLI**.  Kubo is [one of multiple IPFS implementations](.ipfs-implementations.md). It is the oldest IPFS implementation and exposes a CLI (among other things).
```
Comment (lidel) [inline docs/basics/go/command-line.md:8]:
```suggestion
This short guide aims to walk you through the **basics of using IPFS with the Kubo CLI**.  Kubo is [one of multiple IPFS implementations](ipfs-implementations.md). It is the oldest IPFS implementation and exposes a CLI (among other things).
```

**Comment by johnnymatthews** [comment]
It looks like that `tabs` plugin you used has been abandoned. I'm trying out https://github.com/Kaifuny/vuepress-plugin-tabs instead. Gonna convert this PR to a draft since I'm working on it.

**Comment by johnnymatthews** [comment]
Made a few changes to the formatting and copy. I also fixed the tab problem we hit earlier.

Just a few notes @DannyS03:

1. When listing steps, us an ordered list:

    ```markdown
    Lorem ipsum:

    1. First do this.
    1. Then do that.
    1. Finally, do this other thing.
    ```

1. Use markdown syntax for images:

    ```markdown
    ![This is the image alt text.](./images/thingy.png)
    ```

1. Try and use 4 spaces when indenting. The markdown parser specifically looks for 4 spaces, not 2.

    ````markdown
    1. You should obtain an output similar to:

        ```shell
        Added file: test.txt CID(QmYt9ypyGsR1BKdaCGPdwdBgAiuXK5AYN2bGSNZov7YXuk)
        ```

    1. More steps.
    1. Even more steps.
    ````

There's also a few `TODOs` sprinkled through the `js-ipfs.md` file. Ping me again when you've addressed those.

**Comment by DannyS03** [comment]
@johnnymatthews addressed

**Comment by DannyS03** [comment]
With the tabs addition added, will create an issue after this is merged to propose a similar approach for kubo basics

File: docs/reference/js/api.md [hunk 2]
```diff
@@ -5,38 +5,129 @@ description: Developer resources for working in JavaScript with IPFS, the InterP
...
+
+:::
+
+## Packages
+
+> This table is generated using the module [`package-table`](https://www.npmjs.com/package/package-table) with `package-table --data=package-list.json`.
+
+Listing of the main JS packages in the IPFS ecosystem:
+
+| Package | Version | Deps | CI/Travis | Coverage | Lead Maintainer |
```
Comment (2color) [inline docs/reference/js/api.md:81]: We're no longer using Travis. I'd remove this column or replace it with GitHub actions badges, e.g. https://github.com/ipfs/js-ipns/blob/master/README.md?plain=1#L7

File: docs/reference/js/api.md [hunk 3]
```diff
@@ -5,38 +5,129 @@ description: Developer resources for working in JavaScript with IPFS, the InterP
 
 # IPFS in JavaScript
 
-::: callout
-[Explore js-ipfs through interactive coding challenges at ProtoSchool](https://proto.school/course/ipfs)
+## JavaScript libraries
+
+There are two main JavaScript libraries for working with IPFS. Both work in Node.js and modern web browsers.
```
Comment (2color) [inline docs/reference/js/api.md:10]: I'd suggest adding the names of the two libraries here.

File: docs/reference/js/api.md [hunk 4]
```diff
@@ -5,38 +5,129 @@ description: Developer resources for working in JavaScript with IPFS, the InterP
...
+The implementation is split into several modules.
```
Comment (2color) [inline docs/reference/js/api.md:11]: One of the most important details to introduce here is the difference between how `js-ipfs` works in the browser and Node.js.

From a high level:
- Node.js is long-running while the browser is short-lived (and resource-constrained)
- Transport protocols: Node.js supports TCP/UDP/WebSockets while browsers are limited to HTTP, WebSockets , WebRTC. 

For a more details, check out https://www.notion.so/pl-strflt/JS-IP-2022-Roadmap-Draft-e4d3595d5e024179983a8082bc7956cf

File: docs/reference/js/api.md [hunk 5]
```diff
@@ -5,38 +5,129 @@ description: Developer resources for working in JavaScript with IPFS, the InterP
...
 
-::: callout
-[Explore js-ipfs through interactive coding challenges at ProtoSchool](https://proto.school/course/ipfs)
+## JavaScript libraries
+
+There are two main JavaScript libraries for working with IPFS. Both work in Node.js and modern web browsers.
+The implementation is split into several modules. 
+
+::: details Node modules
+
```
Comment (2color) [inline docs/reference/js/api.md:14]: Many of these packages aren't intended for direct consumption. I would move this section lower seeing as this doesn't really help newcomers.

Comment (DannyS03) [reply docs/reference/js/api.md]: addressed

File: docs/reference/js/api.md (hunk 4 above)
Comment (DannyS03) [reply docs/reference/js/api.md]: addressed

File: docs/reference/js/api.md (hunk 3 above)
Comment (DannyS03) [reply docs/reference/js/api.md]: added

**Review by johnnymatthews — APPROVED** [review]

**Pull request merged**